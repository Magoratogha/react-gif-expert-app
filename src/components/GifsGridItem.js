import PropTypes from 'prop-types';

function GifsGridItem({ title, url }) {

    return (
        <div className="gif-card animate__animated animate__bounceIn">
            <img src={ url } alt={ title?.trim() }/>
            <p>{ title?.trim() || 'No title available' }</p>
        </div>
    );
}

GifsGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
};

export default GifsGridItem;