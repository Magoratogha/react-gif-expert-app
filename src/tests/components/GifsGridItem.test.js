import { queryByAttribute, render, screen } from "@testing-library/react";
import GifsGridItem from "../../components/GifsGridItem";

describe('Test in <GifsGridItem />', () => {
    const title = 'Test title';
    const url = 'www.test-url.com';

    test('should render the component properly', () => {
        const { container } = render(<GifsGridItem title={ title } url={ url } />);
        expect(container).toMatchSnapshot();
    });

    test('should to contain a paragraph with the given param title', () => {
        render(<GifsGridItem title={ title } url={ url } />);
        const paragraphElement = screen.queryByText(title.trim(), { selector: 'p' });
        expect(paragraphElement).toBeTruthy();
    });

    test('should to contain an image with the given params title and url', () => {
        render(<GifsGridItem title={ title } url={ url } />);
        const imageElement = screen.queryByAltText(title.trim(), { selector: 'img' });
        expect(imageElement).toBeTruthy();
        expect(imageElement).toHaveAttribute('src', url);
    });

    test('should to contain a div with the "animate__bounceIn" class', () => {
        const { container } = render(<GifsGridItem title={ title } url={ url } />);
        const divElement = queryByAttribute('class', container, 'animate__bounceIn', { exact: false });
        expect(divElement).toBeTruthy();
    });
});