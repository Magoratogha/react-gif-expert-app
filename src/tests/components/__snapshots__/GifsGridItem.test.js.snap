// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifsGridItem /> should render the component properly 1`] = `
<div>
  <div
    class="gif-card animate__animated animate__bounceIn"
  >
    <img
      alt="Test title"
      src="www.test-url.com"
    />
    <p>
      Test title
    </p>
  </div>
</div>
`;
